CMakeLists.txt

```diff
@@ -567,9 +575,13 @@ option(AVIF_ENABLE_GTEST
 )
 
 if(AVIF_BUILD_APPS OR (AVIF_BUILD_TESTS AND AVIF_ENABLE_GTEST))
-    find_package(ZLIB REQUIRED)
-    find_package(PNG 1.6.32 REQUIRED) # 1.6.32 or above for png_get_eXIf_1()/png_set_eXIf_1() and iTXt (for XMP).
-    find_package(JPEG REQUIRED)
+    if(NOT AVIF_LOCAL_ZLIBPNG)
+        find_package(ZLIB REQUIRED)
+        find_package(PNG 1.6.32 REQUIRED) # 1.6.32 or above for png_get_eXIf_1()/png_set_eXIf_1() and iTXt (for XMP).
+    endif()
+    if(NOT AVIF_LOCAL_JPEG)
+        find_package(JPEG REQUIRED)
+    endif()
```

Comment:
There is another `find_package(JPEG REQUIRED)` call at line 151:

```
149     if(BUILD_SHARED_LIBS)
150         # Fix "libyuv.so: undefined reference to `jpeg_read_raw_data'" errors.
151         find_package(JPEG REQUIRED)
152         set(LIBYUV_LIBRARY ${LIBYUV_LIBRARY} ${JPEG_LIBRARY})
153     endif()
```

It should also be guarded by `if(NOT AVIF_LOCAL_JPEG)`.

Reply:
Good catch. It's actually better to check for the definition of JPEG_LIBRARY

Reply:
Checking for `if(NOT AVIF_LOCAL_JPEG)` is equivalent and allows simpler pattern matching with the code here. I leave the choice to you.